### PR TITLE
MultiOp fix conflicts

### PR DESCRIPTION
**Pull request by din-mukhammed**
### Changes
DoMultiOperation will process and commit operations in between if conflicts are found

### Fixes
-

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Review by dabasov — APPROVED** [review]
lgtm

File: zboxcore/sdk/allocation.go [hunk 1]
```diff
@@ -667,63 +667,99 @@ func (a *Allocation) DoMultiOperation(operations []OperationRequest) error {
...
 
-		case constants.FileOperationRename:
-			operation = NewRenameOperation(op.RemotePath, op.DestName, mo.operationMask, mo.maskMU, mo.consensusThresh, mo.fullconsensus, mo.ctx)
+	for i := 0; i < len(operations); {
+		// resetting multi operation and previous paths for every batch
+		mo = MultiOperation{
+			allocationObj: a,
+			operationMask: zboxutil.NewUint128(1).Lsh(uint64(len(a.Blobbers))).Sub64(1),
+			maskMU:        &sync.Mutex{},
+			ctx:           a.ctx,
```
Comment (Roshan-Mehta) [inline zboxcore/sdk/allocation.go]: ctxCncl is missing. Panic issue in system test might be because of this.

**Review by peterlimg — APPROVED** [review]
lgtm

**Comment by Hitenjain14** [comment]
If we want to upload 10 files in a dir ("/a"), it will mark "/a" as true and will have a conflict on every operation and thus will do commit operation 10 times or am I missing something?
cc: @dabasov @Roshan-Mehta

**Comment by din-mukhammed** [comment]
> If we want to upload 10 files in a dir ("/a"), it will mark "/a" as true and will have a conflict on every operation and thus will do commit operation 10 times or am I missing something? cc: @dabasov @Roshan-Mehta

There should be no conflict in such case, fixed it. How about operations on exact same path? Previously there was this check that didn't allow such cases, should we keep this check?
https://github.com/0chain/gosdk/blob/a5173934488d354ee4d009f909958abb2809e843/zboxcore/sdk/allocation.go#L712-L714

**Comment by Hitenjain14** [comment]
> > If we want to upload 10 files in a dir ("/a"), it will mark "/a" as true and will have a conflict on every operation and thus will do commit operation 10 times or am I missing something? cc: @dabasov @Roshan-Mehta
> 
> There should be no conflict in such case, fixed it. How about operations on exact same path? Previously there was this check that didn't allow such cases, should we keep this check?
> 
> https://github.com/0chain/gosdk/blob/a5173934488d354ee4d009f909958abb2809e843/zboxcore/sdk/allocation.go#L712-L714

Marking only the remotePath won't work, we need all the paths from the rootDir and mark them according to the operation.

**Pull request merged**